cmd/package.go

```diff
@@ -153,19 +153,22 @@ func BuildPackage(packageDir string) (string, error) {
 // by comparing previous MD5 hashes to the ones in the current package
 // directory. Only modified files are uploaded and no file deletions are
 // possible at this time.
-func ComposePackage(repo *util.Repo, imageSize int64, updatePackage bool, verbose bool,
-	pullMissing bool, customBoot string, packageDir string, appName string, commandLine string) error {
+func ComposePackage(repo *util.Repo, imageSize int64, updatePackage, verbose, pullMissing bool,
+	packageDir, appName string, bootOpts *SimpleBootOptions) error {
 
 	// Package content should be collected in a subdirectory called mpm-pkg.
 	targetPath := filepath.Join(packageDir, "mpm-pkg")
 	// Remove collected directory afterwards.
 	defer os.RemoveAll(targetPath)
 
 	// Construct final bootcmd for the image.
-	commandLine = constructBootCmdFromArguments(commandLine, customBoot, packageDir)
+	commandLine, err := bootOpts.CalculateCmd()
+	if err != nil {
+		return err
+	}
 
 	// First, collect the contents of the package.
-	if err := CollectPackage(repo, packageDir, pullMissing, customBoot, verbose); err != nil {
+	if err := CollectPackage(repo, packageDir, pullMissing, bootOpts.Boot, verbose); err != nil {
```

Comment:
So, I think I must be missing something very important here - I would appreciate your help understanding this. Here we pass, just the boot, i.e. the name of the run config that we would like to be ran.

If I then go to `CollectPackage` [[link]](https://github.com/miha-plesko/capstan/blob/46022b6d300cdacb242dcd890a1e957523cadc90/cmd/package.go#L227) I see that this `boot` is only needed for `persistBootCmdsIntoFiles` [[link]](https://github.com/miha-plesko/capstan/blob/46022b6d300cdacb242dcd890a1e957523cadc90/cmd/package.go#L618)

Now my problem...

This `persistBootCmdsIntoFiles` begins by creating `cmdConf`, then creates files in `/run/` dir of the unikernel. Afterwards it uses the `customBoot`, to do this:

```
if customBoot != "" {
  cmdConf.ConfigSetDefault = customBoot
}
```

But the `cmdConf` itself doesn't seem to be used at all after that [[link]](https://github.com/miha-plesko/capstan/blob/46022b6d300cdacb242dcd890a1e957523cadc90/cmd/package.go#L655-L662).

Is this related to the fact that we had some issues with the symlinks and decided not to fix them at that time?